In [13]:
import json
import gzip

def load_json_from_file(gt_file_patterns, count=-1):
    data = []
    if gt_file_patterns.endswith('gz'):
        f = gzip.open(gt_file_patterns, 'rt', encoding='utf-8')
    else:
        f = open(gt_file_patterns, 'rt', encoding='utf-8')
    lines = f.readlines()
    for line in lines[0:count]:
        data.append(json.loads(line))
    return data

def write_jsonl(filename: str, content: list[dict], encoding="utf-8"):
    with open(filename, mode="w", encoding=encoding) as fp:
        for entry in content:
            fp.write(f"{json.dumps(entry)}\n")

json_data = load_json_from_file("/dccstor/srosent2/generative/appen/final/longNQ_hf/train/longNQ_train_answerable.jsonl")
json_data[0]

{'id': '8045984229282682032',
 'input': "who sang love the one you're with first",
 'passages': [{'title': "Love the One You're With",
   'text': "`` Love the One You 're With '' is a song by folk rocker Stephen Stills . It was released as the lead single from his debut self - titled studio album in November 1970 . The song , inspired by a remark Stills heard from musician Billy Preston , became his biggest hit single , peaking at No. 14 on the Billboard Hot 100 in early 1971 . David Crosby and Graham Nash , Stills ' fellow members of Crosby , Stills & Nash , provide background vocals on the song . The song was also covered by a number of artists , including The Isley Brothers , Bucks Fizz , and Luther Vandross . ",
   'sentences': ["`` Love the One You 're With '' is a song by folk rocker Stephen Stills .",
    'It was released as the lead single from his debut self - titled studio album in November 1970 .',
    'The song , inspired by a remark Stills heard from musician Billy Preston

{
  "input": "Who has the highest goals in world football?",
  "id": "-7013890438520559398",
  "passages": [
    {
      "title": "List of women's footballers with 100 or more international goals ...",
      "text": "The first player to reach 100 international goals was Italian Elisabetta Vignotto. Abby Wambach"
    }
  ],
  "output": [
    {
      "answer": "Ali Dael has the highest goals in men's world international football with 109 goals. Josef Bican has the highest goals all-time in men's football and Christine Sinclair has the highest goals in women's world international football.",
      "meta": {}
    },
    {
      "answer": "The players with the highest all-time goals and highest men's and women's international football goals differ.",
      "meta": {}
    }
  ]
}

{
  "id": "8045984229282682032",
  "input": "who sang love the one you're with first",
  "passages": [
    {
      "title": "Love the One You're With",
      "text": "`` Love the One You 're With '' is a song by folk rocker Stephen Stills . It was released as the lead single from his debut self - titled studio album in November 1970 . The song , inspired by a remark Stills heard from musician Billy Preston , became his biggest hit single , peaking at No. 14 on the Billboard Hot 100 in early 1971 . David Crosby and Graham Nash , Stills ' fellow members of Crosby , Stills & Nash , provide background vocals on the song . The song was also covered by a number of artists , including The Isley Brothers , Bucks Fizz , and Luther Vandross . "
    }
  ],
  "output": [
    {
      "answer": "`` Love the One You 're With '' is a song by folk rocker Stephen Stills . David Crosby and Graham Nash , Stills ' fellow members of Crosby , Stills & Nash , provide background vocals on the song . The song was also covered by a number of artists , including The Isley Brothers , Bucks Fizz , and Luther Vandross .",
      "meta": {}
    }
  ]
}


In [14]:
json_data_nometa = []

for example in json_data:
    del example['passages'][0]['sentences']

    for output in example['output']:
        del output['selected_sentences']
        output['meta'] = {}
    json_data_nometa.append(example)

write_jsonl("/dccstor/srosent2/generative/external_datasets/asqa+longnq/train_longnq.jsonl", json_data_nometa)

In [20]:
data = load_json_from_file("/dccstor/srosent2/generative/external_datasets/asqa+longnq/train_longnq.jsonl")
data = load_json_from_file("/dccstor/srosent2/generative/external_datasets/asqa+longnq/train.jsonl")

for example in data:
    if 'id' not in example or type(example['id']) is not str:
        print("missing id")
    if 'input' not in example:
        print("missing question")
    if 'output' not in example:
        print('missing output')
    if 'passages' not in example:
        print('missing passages')